share/rocm/cmake/ROCMSphinxDoc.cmake

```diff
@@ -1,63 +1,89 @@
 # ######################################################################################################################
-# Copyright (C) 2021 Advanced Micro Devices, Inc.
+# Copyright (C) 2023 Advanced Micro Devices, Inc.
 # ######################################################################################################################
 
+include_guard(GLOBAL)
+
 include(CMakeParseArguments)
 include(ROCMDocs)
 
 find_program(
     SPHINX_EXECUTABLE
     NAMES sphinx-build
-    HINTS $ENV{SPHINX_DIR}
+    HINTS "$ENV{SPHINX_DIR}"
     PATH_SUFFIXES bin
-    DOC "Sphinx documentation generator")
-
-mark_as_advanced(SPHINX_EXECUTABLE)
+    DOC "Sphinx documentation generator"
+    REQUIRED)
 
-set(BINARY_BUILD_DIR "${CMAKE_CURRENT_BINARY_DIR}/sphinx/_build")
+find_program(
+    DOXYGEN_EXECUTABLE
```

Comment:
Should `DOXYGEN_EXECUTABLE` also be marked as advanced?

Reply:
It probably should. Added.